<a href="https://colab.research.google.com/github/HKyeol/Python_study/blob/master/201216%EC%9B%B9%EC%8A%A4%ED%81%AC%EB%A1%A4%EB%A7%81.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

* 빅데이터 수집을 위해 반드시 알아야할 필수 기술? 웹스크롤링

* 웹스크롤링을 위해서 사용한 파이썬 모듈? beautiful soup 모듈  

* 웹스크롤링을 편하게 잘 하기 위한 비법?  
크롬의 개발자 모드(F12)로 들어가서 스크롤링 하기 원하는 지점의 html 클래스 이름을 빠르게 찾을 수 있습니다.  
이전엔 ctrl + U 로 소스보기를 한 다음에 쭈우욱 찾아야 했다...  


# 142. 웹스크롤링 실전 1단계 (ebs 레이디 버그 게시판)  

우리회사의 신제품이 출시 되었을 때 그 제품에 대한 사람들의 반응을 데이터 분석을 하고자 할 때 ***웹 스크롤링 + 데이터 시각화***를 이용하면 됩니다.  

예제1. ebs 레이디버그 시청자 게시판의 url을 가지고 직접 html 문서를 내려받을 수 있도록 코드를 구현  
(어제는 우리가 ctrl + s를 누르고 웹페이지를 우리 피씨에 저장하고 구현을 했는데 오늘은 그렇게 하지 않고 웹에서 바로 html 문서를 내려 받을 수 있도록 구현

In [ ]:
from bs4 import BeautifulSoup
import urllib.request # 웹상의 url을 파이썬이 인식할 수 있도록 해주는 모듈

list_url = "http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?hmpMnuId=106"
url = urllib.request.Request(list_url) # 사람이 알아볼 수 있는 위의 url을 파이썬이 알아볼 수 있도록 변환 
                                       # 첫번째 변환 작업
print(url) # <urllib.request.Request object at 0x7efce93a7780>
result = urllib.request.urlopen(url).read().decode("utf-8") # decode = 사람이 볼 수 있는 언어로 변환 
                                       # 두번째 변환 작업. 위의 url의 html 문서들을 result 변수에 담았다.
print(result)   # 위의 url의 html 전체 문서가 다 출력되고 있음

이제 파싱을 해보자.  

예제2. 위에서 긁어온 html 코드를 BeautifulSoup의 함수를 이용해서 웹 스크롤링을 할 수 있도록 BeautifulSoup을 쓸 수 있게 파싱하시오

In [ ]:
from bs4 import BeautifulSoup
import urllib.request # 웹상의 rul을 파이썬이 인식할 수 있도록 해주는 모듈

list_url = "http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?hmpMnuId=106"
url = urllib.request.Request(list_url) # 사람이 알아볼 수 있는 위의 url을 파이썬이 알아볼 수 있도록 변환 
                                       # 첫번째 변환 작업
result = urllib.request.urlopen(url).read().decode("utf-8") # decode = 사람이 볼 수 있는 언어로 변환 
                                       # 두번째 변환 작업. 위의 url의 html 문서들을 result 변수에 담았다.
soup = BeautifulSoup(result, "html.parser") # 뷰티풀숲을 이용할 수 있게 파싱하는 거
print(soup)

예제3. 지금 페이지의 시청자 게시판의 글 내용에 해당하는 부분의 태그와 클래스 이름을 알아내시오.  
재미있다랑 제일 가까운 클래스 이름을 확인하자. con이다.
```
< p class_ = "con" > 재미있 다 </p>
```

예제4. p 태그중에 class가 con에 해당하는 부분을 스크롤링 하시오.
```
('p',class_ ='con')
p 태그 중에서, class가 con인 애를 스크롤링 해라
```
p 빼면 다른값도 나옴  
find_all 말고 find만 하면 맨 처음 하나만 가져온다.

In [ ]:
list_url = "http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?hmpMnuId=106"
url = urllib.request.Request(list_url)
result = urllib.request.urlopen(url).read().decode("utf-8")
soup = BeautifulSoup(result, "html.parser")
result2 = soup.find_all('p',class_ ='con') # find랑 find_all 비교
print(result2)

예제5. 위의 결과에서 html 문서 말고 한글 텍스트만 가져오시오  
위의 result2는 리스트 입니다.
```
</p>  ] 이렇게 리스트로 담겨있다.
```
'+ 예제6. 위에서 출력되고 있는 텍스트들이 좀 더 깔끔하게 나오게 하시오!

In [ ]:
list_url = "http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?hmpMnuId=106"
url = urllib.request.Request(list_url)
result = urllib.request.urlopen(url).read().decode("utf-8")
soup = BeautifulSoup(result, "html.parser")
result2 = soup.find_all('p',class_ ='con') # find랑 find_all 비교
#print(result.get_text().strip()) # find 일 때는 이걸로
for i in result2:
    print(i.get_text().strip()) # 내 답
for i in result2:
    print(i.get_text(" ", strip=True)) # 정답! 와!

예제7. 위에서 출력되고 있는 텍스트들이 params라는 비어있는 리스트에 담기게 하시오!

In [ ]:
list_url = "http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?hmpMnuId=106"
url = urllib.request.Request(list_url)
result = urllib.request.urlopen(url).read().decode("utf-8")
soup = BeautifulSoup(result, "html.parser")
result2 = soup.find_all('p',class_ ='con') # find랑 find_all 비교

params = []
for i in result2:
    params.append(i.get_text(" ", strip=True))
params

예제8. 게시글을 올린 날짜를 스크롤링하기 위해서 게시글 날짜가 있는 html 문서의 태그 이름과 클래스 이름을 확인하시오!  
```
태그이름은 span이고 클래스 이름은 date 입니다.
```
예제9. 기존 코드에 위의 날짜를 스크롤링하는 코드를 추가하고 위의 날짜를 모두 스크롤링해서 param2 라는 리스트에 담으시오.

In [ ]:
# 1. 웹에서 html 문서를 가져와 beautifulsoup으로 파싱
from bs4 import BeautifulSoup
import urllib.request # 웹상의 url을 파이썬이 인식할 수 있도록 해주는 모듈

list_url = "http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?hmpMnuId=106"
url = urllib.request.Request(list_url)
result = urllib.request.urlopen(url).read().decode("utf-8")
soup = BeautifulSoup(result, "html.parser")

# 2. 시청자 게시판의 날짜와 본문 내용을 가져옵니다.
result1 = soup.find_all('p',class_ ='con') # find랑 find_all 비교
result2 = soup.find_all('span',class_ = 'date')

# 3. 시청자 게시판의 날짜와 본문을 params1와 rapams2 리스트에 담습니다.
params1 = []
params2 = []
for i,k in zip(result1,result2):
    params1.append(i.get_text(" ", strip=True))
    params2.append(k.get_text())
print(params1)
print(params2)


['재미있 다', '어제 흐름상 종영 같지 않았는데 갑자기 끝이네요??', '〈미라큘러스: 레이디버그와 블랙캣〉종영 안내!\r\n\r\n\r\n\r\n〈미라큘러스: 레이디버그와 블랙캣〉-시즌3가 \r\n\r\n방송 종료되었습니다, 그동안 시청해주셔서 감사드립니다!\r\n\r\n2021년 새로운 시즌으로 여러분을 찾아뵙도록 노력하겠습니다!\r\n\r\n고맙습니다~', '아이들 때문에 보게 되었는데 어른인 제가 더 좋아하는 프로그램이 되었네요 아이들이랑 볼 때 너무 신나요 함께 볼 수 있는 보기 드문 애니매이션이라 더 좋네요 시즌 4도 너무 너무 기대되고 기다리고 있습니다', '2021년이 빨리되서 레이디버그 시즌4를 보면 좋겠네요~', '레이디버그... 제가 어렸을때도 너~~무 좋아 하던 방송 이에요!!!!!!!', '이렇게 재미있는프로는 이 세상에 하나뿐이예요 아 그리고 뉴욕편은 언제 올려주시나요?', '너모 재밌어요!!! 저 시즌 1부터 다 본방으로 챙겨본 미라큘러 입니다ㅋㅋㅋ 진짜  재밌구요 팬카페랑 이런데도 다 가입해있어요... 방도 레이디버그 사진으로 덕-질☆', '완전 재밌다', '근뎅.....오즈 긑나서....레이디버그가 하는거잖아여....ㅠ', '우왕!', '넘재밌어요.매일하면 좋겠어요', '재밌어요 완전 꿀잼이요 사랑합니다', 'ㅎㅎ', '진짜여???', '보니하니,레이디버그,형사 가제트,오즈,삐삐에 있었어요!', '오늘부터 미라큘러스 시즌3 다시 방영한다네요!^^\r\n목,금 저녁 7시', '전 세매큐,픽시,가제트형사,오:마법을 찾아서,히어로써클,로빈후드에 있어여!ㅎㅎ즈', '벌써 가을 인길요~', '가을이 빨리 되면 좋겠어요']
['2020.12.11 19:52', '2020.12.11 19:06', '2020.12.11 01:35', '2020.12.10 12:43', '2020.10.17 15:30', '2020.10.16 12:25', '2020.10.15 21:04', '2020.09.29 14:46', '2020.09.29 11:30

예제10. 위의 날짜와 본문 내용이 아래와 같이 출력되게 하시오!
```
2020.12.11 19:52 재미있 다
..
.
```

In [ ]:
# 1. 웹에서 html 문서를 가져와 beautifulsoup으로 파싱
from bs4 import BeautifulSoup
import urllib.request # 웹상의 url을 파이썬이 인식할 수 있도록 해주는 모듈

list_url = "http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?hmpMnuId=106"
url = urllib.request.Request(list_url)
result = urllib.request.urlopen(url).read().decode("utf-8")
soup = BeautifulSoup(result, "html.parser")

# 2. 시청자 게시판의 날짜와 본문 내용을 가져옵니다.
result1 = soup.find_all('p',class_ ='con') # find랑 find_all 비교
result2 = soup.find_all('span',class_ = 'date')


# 3. 시청자 게시판의 날짜와 본문을 params1와 rapams2 리스트에 담습니다.
params1 = []
params2 = []
for i,k in zip(result1,result2):
    params1.append(i.get_text(" ", strip=True))
    params2.append(k.get_text())

# 4. 날짜와 본문을 같이 출력합니다.
for j, h in zip(params2,params1):
    print(j + ' ' + h)

예제11.(중요) 게시글이 굉장히 많다. 레이디 버그 게시판 전체의 글을 다 스크롤링해서 예제 10번 결과처럼 출력되게 하시오!
```
URL을 긁어서 보다보면 패턴을 발견할 수 있다.
http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?c.page=1&hmpMnuId=106&searchKeywordValue=0&bbsId=10059819&searchKeyword=&searchCondition=&searchConditionValue=0&
http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?c.page=2&hmpMnuId=106&searchKeywordValue=0&bbsId=10059819&searchKeyword=&searchCondition=&searchConditionValue=0&
http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?c.page=3&hmpMnuId=106&searchKeywordValue=0&bbsId=10059819&searchKeyword=&searchCondition=&searchConditionValue=0&
```
위의 url을 가만히 살펴보면 페이지 번호만 다르고 나머지는 다 같다는 것을 확인할 수 있습니다.  
예제12. 아래의 결과를 출력하는데 페이지 번호가 1번 부터 22번까지 변경되어서 출력되게하시오 !  
예제13. 예제12번 코드를 예제10번 코드에 적용해서 레이디 버그 전체 게시판의 글들이 날짜와 함께 출력되게 하시오

In [ ]:
# 1. 웹에서 html 문서를 가져와 beautifulsoup으로 파싱
from bs4 import BeautifulSoup
import urllib.request # 웹상의 url을 파이썬이 인식할 수 있도록 해주는 모듈
page_num = 0
for i in range(1,23):
    page_num += 1
    list_url = f"http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?c.page={page_num}&hmpMnuId=106&searchKeywordValue=0&bbsId=10059819&searchKeyword=&searchCondition=&searchConditionValue=0&"
    url = urllib.request.Request(list_url)
    result = urllib.request.urlopen(url).read().decode("utf-8")
    soup = BeautifulSoup(result, "html.parser")

    # 2. 시청자 게시판의 날짜와 본문 내용을 가져옵니다.
    result1 = soup.find_all('p',class_ ='con') # find랑 find_all 비교
    result2 = soup.find_all('span',class_ = 'date')


    # 3. 시청자 게시판의 날짜와 본문을 params1와 rapams2 리스트에 담습니다.
    params1 = []
    params2 = []
    for i,k in zip(result1,result2):
        params1.append(i.get_text(" ", strip=True))
        params2.append(k.get_text())

    # 4. 날짜와 본문을 같이 출력합니다.
    for j, h in zip(params2,params1):
        print(j + ' ' + h)

설명: 위의 params1과 params2가 for문 안쪽에 있기 때문에 페이지 번호가 돌 때 마다 params 리스트 안의 내용이 새로운 날짜와 내용으로 변경되게 됩니다.  

예제14. params1과 params2에 레이디 버그 시청자 게시판의 모든 게시날짜와 본문 내용이 들어가게 코드를 수정하시오!

In [ ]:
# 0. 웹에서 html 문서를 가져와 beautifulsoup으로 파싱
from bs4 import BeautifulSoup
import urllib.request # 웹상의 url을 파이썬이 인식할 수 있도록 해주는 모듈
page_num = 0
# 1. 레이디 버그 게시판 게시날짜와 게시글 전체를 다 담은 리스트 2개를 만듭니다.
params1 = []
params2 = []

for i in range(1,23):
    page_num += 1
    list_url = f"http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?c.page={page_num}&hmpMnuId=106&searchKeywordValue=0&bbsId=10059819&searchKeyword=&searchCondition=&searchConditionValue=0&"
    url = urllib.request.Request(list_url)
    result = urllib.request.urlopen(url).read().decode("utf-8")
    soup = BeautifulSoup(result, "html.parser")

    # 2. 시청자 게시판의 날짜와 본문 내용을 가져옵니다.
    result1 = soup.find_all('p',class_ ='con') # find랑 find_all 비교
    result2 = soup.find_all('span',class_ = 'date')


    # 3. 시청자 게시판의 날짜와 본문을 params1와 rapams2 리스트에 담습니다.

    for i,k in zip(result1,result2):
        params1.append(i.get_text(" ", strip=True))
        params2.append(k.get_text())

# 4. 날짜와 본문을 같이 출력합니다.
for j, h in zip(params2,params1):
    print(j + ' ' + h)

설명: 예제13번은 레이디 버그 게시판 전체글을 화면에 출력하는 코드이고 예제14는 레이디 버그 게시판 전체글을 params1과 params2 리스트에 모두 담고 화면에 출력하는 코드

문제415. 점심시간 문제) 레이디 버그 게시판의 전체 게시글은 총 몇건인가? 코드와 답을 올리시오.

In [ ]:
# 0. 웹에서 html 문서를 가져와 beautifulsoup으로 파싱
from bs4 import BeautifulSoup
import urllib.request # 웹상의 url을 파이썬이 인식할 수 있도록 해주는 모듈
page_num = 0

cnt = 0
for i in range(1,23):
    page_num += 1
    list_url = f"http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?c.page={page_num}&hmpMnuId=106&searchKeywordValue=0&bbsId=10059819&searchKeyword=&searchCondition=&searchConditionValue=0&"
    url = urllib.request.Request(list_url)
    result = urllib.request.urlopen(url).read().decode("utf-8")
    soup = BeautifulSoup(result, "html.parser")

    result2 = soup.find_all('span',class_ = 'date')

    cnt += len(result2)
print(cnt)

429


문제416. 게시글 429건 전체를 c드라이브 밑에 mytext34.txt라는 이름으로 저장하시오  
* 텍스트 파일을 저장하는 파이썬 코드


In [ ]:
text = 'assdfasd'
f = open(':c\\data\mytext32.txt','w',encoding='UTF8')
f.writ(text)
f.close


In [ ]:
# 0. 웹에서 html 문서를 가져와 beautifulsoup으로 파싱
from bs4 import BeautifulSoup
import urllib.request # 웹상의 url을 파이썬이 인식할 수 있도록 해주는 모듈
page_num = 0
# 1. 레이디 버그 게시판 게시날짜와 게시글 전체를 다 담은 리스트 2개를 만듭니다.
params1 = []
params2 = []

for i in range(1,23):
    page_num += 1
    list_url = f"http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?c.page={page_num}&hmpMnuId=106&searchKeywordValue=0&bbsId=10059819&searchKeyword=&searchCondition=&searchConditionValue=0&"
    url = urllib.request.Request(list_url)
    result = urllib.request.urlopen(url).read().decode("utf-8")
    soup = BeautifulSoup(result, "html.parser")

    # 2. 시청자 게시판의 날짜와 본문 내용을 가져옵니다.
    result1 = soup.find_all('p',class_ ='con') # find랑 find_all 비교
    result2 = soup.find_all('span',class_ = 'date')


    # 3. 시청자 게시판의 날짜와 본문을 params1와 rapams2 리스트에 담습니다.

    for i,k in zip(result1,result2):
        params1.append(i.get_text(" ", strip=True))
        params2.append(k.get_text())

# 4. 날짜와 본문을 같이 출력합니다.
f = open('c:\\data\\mydata34.txt','w') # w는 write 입니다.

for j, h in zip(params2,params1):
    f.write(j + ' ' + h + '\n')
f.close()

다른 사이트를 연습해보러 가자  

# 143. 웹스크롤링 실전2 (중앙일보사)

중앙일보사 홈페이지에서 인공지능으로 검색을 했을 때 나오는 기사들을 전부 웹스크롤링 한다.  

예제1. 중앙일보사 홈페이지에서 인공지능으로 검색했을 때 나오는 url을 가져오시오.
```
https://news.joins.com/Search/JoongangNews?page=1&Keyword=%EC%9D%B8%EA%B3%B5%EC%A7%80%EB%8A%A5&SortType=New&SearchCategoryType=JoongangNews
```


예제2. 위의 사이트에서 보이는 상세기사를 클릭하고 그 기사의 url을 복사하시오.
```
https://news.joins.com/article/23947044
https://news.joins.com/article/23946979
https://news.joins.com/article/23946876
```
검색했을 때 나오는 url은 페이지 번호처럼 1234 순이 아니다.  

예제3. 인공지능으로 검색했을 때 나오는 기사들의 url을 스크롤링 하시오.


In [ ]:
#1. 웹스크롤링에 필요한 모듈을 임포트합니다.
from bs4 import BeautifulSoup
import urllib.request

#2. 중앙일보에서 인공지능으로 검색했을 때 나오는 첫페이지의 html 코드를
# beautifulSoup에서 이용할 수 있도록 파싱합니다.
page_num = 1
list_url = f"https://news.joins.com/Search/JoongangNews?page={page_num}&Keyword=%EC%9D%B8%EA%B3%B5%EC%A7%80%EB%8A%A5&SortType=New&SearchCategoryType=JoongangNews"
url = urllib.request.Request(list_url)
result = urllib.request.urlopen(url).read().decode("utf-8")
soup = BeautifulSoup(result, "html.parser")

#3. 상세기사 url을 가져올 수 있도록 태그와 클래스를 찾습니다.
# 찾아보니 태그는 h2였고 클래스 이름은 headline mg 입니다.
result1 = soup.find_all('h2',class_ ='headline mg') # find랑 find_all 비교

#4. 위의 result1은 리스트 이므로 for loop 문을 이용해서 리스트에 있는 요소를
# 하나씩 빼내면서 상세 url에 해당하는 href의 값을 얻어냅니다.

for i in result1:   # result1 리스트의 요소를 하나씩 빼내는 코드
    for k in i:     # h2 태그안의 a 태그의 html 코드를 가져오기 위함
        print(k.get('href'))    # 이거 기억하자

https://news.joins.com/article/23947044
https://news.joins.com/article/23946979
https://news.joins.com/article/23946876
https://news.joins.com/article/23946841
https://news.joins.com/article/23946757
https://news.joins.com/article/23946642
https://news.joins.com/article/23946268
https://news.joins.com/article/23946216
https://news.joins.com/article/23946191
https://news.joins.com/article/23946032


예제4. 위의 상세 기사 url을 params라는 비어 있는 리스트에 다 append 시키시오.

In [ ]:
#1. 웹스크롤링에 필요한 모듈을 임포트합니다.
from bs4 import BeautifulSoup
import urllib.request

#2. 중앙일보에서 인공지능으로 검색했을 때 나오는 첫페이지의 html 코드를
# beautifulSoup에서 이용할 수 있도록 파싱합니다.
page_num = 1
list_url = f"https://news.joins.com/Search/JoongangNews?page={page_num}&Keyword=%EC%9D%B8%EA%B3%B5%EC%A7%80%EB%8A%A5&SortType=New&SearchCategoryType=JoongangNews"
url = urllib.request.Request(list_url)
result = urllib.request.urlopen(url).read().decode("utf-8")
soup = BeautifulSoup(result, "html.parser")

#3. 상세기사 url을 가져올 수 있도록 태그와 클래스를 찾습니다.
# 찾아보니 태그는 h2였고 클래스 이름은 headline mg 입니다.
result1 = soup.find_all('h2',class_ ='headline mg') # find랑 find_all 비교

#4. 위의 result1은 리스트 이므로 for loop 문을 이용해서 리스트에 있는 요소를
# 하나씩 빼내면서 상세 url에 해당하는 href의 값을 얻어냅니다.
params = []
for i in result1:   # result1 리스트의 요소를 하나씩 빼내는 코드
    for k in i:     # h2 태그안의 a 태그의 html 코드를 가져오기 위함
        params.append(k.get('href'))    # 이거 기억하자
params

['https://news.joins.com/article/23947044',
 'https://news.joins.com/article/23946979',
 'https://news.joins.com/article/23946876',
 'https://news.joins.com/article/23946841',
 'https://news.joins.com/article/23946757',
 'https://news.joins.com/article/23946642',
 'https://news.joins.com/article/23946268',
 'https://news.joins.com/article/23946216',
 'https://news.joins.com/article/23946191',
 'https://news.joins.com/article/23946032']

예제5. 상세기사 url 중에 하나를 복사해 오고 그 상세기사 url의 웹페이지로 접속해서 그 본문기사의 태그 이름과 클래스 이름이 무엇인지 확인하시오.  
답: 캐그 이름은 div 태그이고 클래스 이름은 "article_body mg fs4" 입니다.  

예제6. 위의 상세기사의 본문을 스크롤링해서 출력하시오.

In [ ]:
from bs4 import BeautifulSoup
import urllib.request

list_url = "https://news.joins.com/article/23947044"
url = urllib.request.Request(list_url)
result = urllib.request.urlopen(url).read().decode("UTF-8")
soup = BeautifulSoup(result,"html.parser")

result1 = soup.find_all('div',class_ = "article_body mg fs4")
params = []
for i in result1: # result1 리스트의 요소를 하나씩 빼내는 코드
    params.append(i.get_text(" ", strip = True))
print(params)

['김재철 동원그룹 명예회장이 16일 대전 KAIST 본원에서 열린 \'AI 발전기금 기부 약정식\'에서 500억원 기부를 발표했다. 프리랜서 김성태 “세계 역사는 대항해시대를 거치면서 패권의 큰 흐름이 바뀌었다. 이제 과거 모든 변화를 아우르는 21세기 4차 산업혁명의 시대에는 인공지능(AI)을 지배하는 자가 세계패권을 쥐게 된다.” “세계 역사는 대항해시대 이후 패권 바뀌어 21세기는 AI 지배하는 자가 세계패권 쥔다” KAIST 올 한해 총 1474억원 모아, 역대 최대 ‘마도로스’김재철(84) 동원그룹 명예회장이 AI 인재육성과 연구개발을 위해 \xa0KAIST에 500억원을 기부했다. \xa0KAIST는 16일 대전 본원에서 동원그룹 김재철 명예회장의 ‘AI 발전기금 기부 약정식’을 했다고 밝혔다. 약정식에는 신성철 총장 등 KAIST 주요 보직 교수들과 김 명예회장, 김 회장의 두 아들 김남구 한국금융지주 회장, 김남정 동원그룹 부회장 등이 참석했다. 김 명예회장은 약정식에서 “AI 물결이 대항해시대와 1ㆍ2ㆍ3차 산업혁명 이상으로 우리의 삶을 바꾸는 큰 변화를 이끌 것”이라며 “오늘 이 자리는 대한민국이 AI 선진국으로 나아가는 출정식이 될 것”이라고 말했다. 김 명예회장은 또 “위대한 잠재력을 가진 우리 국민이 국력을 모아 경쟁에 나서면 AI 선진국이 될 수 있다”며 “과학영재와 우수한 교수진이 집결해 있는 KAIST가 선두주자로서 우리나라 AI 개발 속도를 촉진하는 플래그십 역할을 할 것”이라고 덧붙였다. KAIST AI대학원은 지난해 3월 국내 대학 중 최초로 과학기술정보통신부의 AI대학원 지원사업에 선정됐으며, 5개월 뒤인 8월 문을 열었다. 지난달 말 현재 구글ㆍIBM왓슨ㆍ마이크로소프트 등 글로벌 정보기술(IT) 기업의 AI연구소 출신 전임교수 13명과 겸임교수 8명 등 21명으로 교수진을 구성하고 있다. 내년 3월부터는 단계적으로 캠퍼스를 대전 본원에서 서울 홍릉으로 이전할 계획이다. 또 2023년에는 서울 양재동 ‘양재 연구개발 혁신지

예제7. 상세기사 url을 가져와서 params 리스트에 넣고 출력하는 예제4번 코드를 함수로 생성하시오!
```
print(j_scroll()) 딱 치면

['https://news.joins.com/article/23947044',
 'https://news.joins.com/article/23946979',
 'https://news.joins.com/article/23946876',
 'https://news.joins.com/article/23946841',
 'https://news.joins.com/article/23946757',
 'https://news.joins.com/article/23946642',
 'https://news.joins.com/article/23946268',
 'https://news.joins.com/article/23946216',
 'https://news.joins.com/article/23946191',
 'https://news.joins.com/article/23946032']

 나오게 하시오
 ```

In [ ]:
def j_scroll(page_num): # 중앙일보 페이지 받아옴
    #1. 웹스크롤링에 필요한 모듈을 임포트합니다.
    from bs4 import BeautifulSoup
    import urllib.request

    #2. 중앙일보에서 인공지능으로 검색했을 때 나오는 첫페이지의 html 코드를
    # beautifulSoup에서 이용할 수 있도록 파싱합니다.
    list_url = f"https://news.joins.com/Search/JoongangNews?page={page_num}&Keyword=%EC%9D%B8%EA%B3%B5%EC%A7%80%EB%8A%A5&SortType=New&SearchCategoryType=JoongangNews"
    url = urllib.request.Request(list_url)
    result = urllib.request.urlopen(url).read().decode("utf-8")
    soup = BeautifulSoup(result, "html.parser")

    #3. 상세기사 url을 가져올 수 있도록 태그와 클래스를 찾습니다.
    # 찾아보니 태그는 h2였고 클래스 이름은 headline mg 입니다.
    result1 = soup.find_all('h2',class_ ='headline mg') # find랑 find_all 비교

    #4. 위의 result1은 리스트 이므로 for loop 문을 이용해서 리스트에 있는 요소를
    # 하나씩 빼내면서 상세 url에 해당하는 href의 값을 얻어냅니다.
    params = []
    for i in result1:   # result1 리스트의 요소를 하나씩 빼내는 코드
        for k in i:     # h2 태그안의 a 태그의 html 코드를 가져오기 위함
            params.append(k.get('href'))    # 이거 기억하자
    return params

In [ ]:
print(j_scroll(2))

['https://news.joins.com/article/23945689', 'https://news.joins.com/article/23945435', 'https://news.joins.com/article/23945158', 'https://news.joins.com/article/23944857', 'https://news.joins.com/article/23944649', 'https://news.joins.com/article/23944441', 'https://news.joins.com/article/23943925', 'https://news.joins.com/article/23943585', 'https://news.joins.com/article/23943561', 'https://news.joins.com/article/23943439']


예제8. 상세기사 url로 본문 기사를 스크롤링해서 리스트에 담았던 예제6번을 j_detail_scroll() 함수로 생성하시오!

In [ ]:
def  j_detail_scroll(num): # url 주소 받아옴
    list_url = j_scroll(num)

    params2 = []
    for i in list_url:
        url = urllib.request.Request(i)
        result = urllib.request.urlopen(url).read().decode("UTF-8")
        soup = BeautifulSoup(result,"html.parser")

        result1 = soup.find_all('div',class_ = "article_body mg fs4")

        for i in result1: # result1 리스트의 요소를 하나씩 빼내는 코드
            params2.append(i.get_text(" ", strip = True))
    return params2

In [ ]:
print(j_detail_scroll('https://news.joins.com/article/23945435'))

['인공지능(AI)이 질병을 미리 알려주고, 평균 신약 개발 기간을 10년에서 3년으로 줄여주는 세상. 생각보다 머지 않은 미래다. 국내에서도 의료 AI 기업들이 선전하고 있다. 저명한 학술지에 소개될 만큼 기술력도 인정받고 있다. 하지만 글로벌 수준의 대규모 투자 유치 소식은 아직이다. 왜일까. 의사와 AI가 함께 질환을 판독하는 시대 [사진 셔터스톡] 무슨 일이야 딥러닝 기반 국내 의료 AI 스타트업 의 성장세가 가파르다. 눈에 띄는 곳은 \' 루닛 \'과 \' 뷰노 \'. 엑스레이·MRI·CT 영상 등을 보고 질환 신호를 읽어내는 AI를 만든다. 이들은 이달 초 보건복지부 인증 \'30개 혁신형 의료기기 기업\'에 나란히 꼽혔다. · 루닛의 폐암·유방암 진단 AI는 서울대병원·아산병원 등 국내 상위 10개 대형병원 중 7개가 사용한다. 정확도는 97~99%. 김은경 용인세브란스병원 영상의학과 교수는 "1년 전 루닛 AI를 도입한 뒤 판독 수준이 균일하게 상향 평준화됐다. \'경험 많은 베테랑 의사\'급 이다"며 "의사의 영상 판독 시간도 평균 10시간에서 8시간으로 20% 줄어 환자 상담 등에 쓸 수 있는 시간이 많아졌다"고 말했다. 루닛이 유치한 누적 투자액은 600억원. · 뷰노는 뼈 나이 판독 AI, 치매 진단 AI, 흉부·망막질환 진단 AI 등을 만든다. 자체 연구 결과 의사의 판독 시간을 40~50% 단축한다. 전국 의료기관 200여 곳이 뷰노의 AI를 쓴다. 누적 투자유치 250억원. · 두 기업 모두 글로벌 기업에 기술을 공급하고 있다. 지난 6월 루닛은 세계 1위 영상장비업체 GE헬스케어 와, 뷰노는 일본 최대 의료정보 플랫폼 M3 와 솔루션 판권 계약을 맺었다. GE와 M3의 의료 장비에 루닛과 뷰노의 기술이 각각 탑재된다는 뜻. · 의료 현장에서도 긍정적 평가 가 나온다. 한부경 삼성서울병원 영상의학과 교수는 "기존에 쓰던 캐드(CAD·20여년 전 나온 의료영상 분석 소프트웨어)는 정확하게 진단을 잘 못해 \'양치기 소년\' 취급을 받았

예제9. 지금 만든 j_detail_scroll() 함수는 상세기사 딱 한개의 본문을 출력하는 함수인데 이 j_detail_scroll() 함수에 j_scroll() 함수를 실행했을 때 나오는 상세 url 여러개를 제공할 수 있도록 코드를 수정하시오!

In [ ]:
print(j_detail_scroll(1))

['명지대학교(총장 유병진) ‘명지스마트팩토리(MSF) 개소식’이 12월 16일(수) 오전 10시 30분 명지대학교 제1공학관 B1층 MSF(Y024호)에서 개최됐다. 명지스마트팩토리는 4차산업혁명의 미래형 산업변화에 대응하기 위해서 스마트 임베디드 플랫폼 사업단(SEP)에서 2019년 3월부터 교육부 대학혁신지원사업의 일환으로 기획 및 구축해왔다. 이번 행사는 코로나19의 확산을 방지하기 위해 유병진 총장을 비롯하여 소수의 관계자만 참석하였으며, 정부의 방역지침 및 사회적 거리두기 지침을 준수하며 온라인 커뮤니케이션 플랫폼 줌(Zoom)으로 진행되었다. ‘명지스마트팩토리(MSF) 개소식’은 개회, 기도(김진옥 교목), 참석자 소개, 축사(유병진 총장), MSF 구축 성과 보고, 테이프 커팅. MSF 시연 및 질의응답, 맺음말, \xa0폐회, 단체 기념촬영 순으로 진행되었다. 명지스마트팩토리(MSF) 구축 사업에는 3년간 약 11억 원의 재정지원 사업비가 투입되었으며, 스마트팩토리 설계와 시설공사 및 기자재 구매를 진행하여 4차산업혁명 시대에 요구되는 첨단기술이 적용된 교육용 스마트팩토리를 구축하였다. 명지 스마트 팩토리에서는 지능형 임베디드 기전공학(스마트로봇, 스마트자동차, 인공지능 소프트웨어) 관련 실무교육을 강화하고 4차산업 혁명시대를 선도하는 명지대학교의 브랜드로서의 역할을 강화한다. 유병진 총장은 축사에서 “오늘 개소하는 명지스마트팩토리는 4차산업혁명의 핵심인 스마트자동차와 스마트로봇으로 구성된 교육용 스마트팩토리로, 4차산업혁명의 신기술에 대한 요소 기술과 시스템 통합기술을 학습할 수 있는 시설이다”며 “명지스마트팩토리는 향후 4차산업혁명 시대를 선도하는 유능한 융합인재를 육성할 수 있는 훌륭한 교육의 장이 될 것이다. 첨단 기술들이 어떻게 작동하고, 어떻게 연결되는지 몸소 체험하며 훌륭한 인재로 발돋움하게 될 것”이라고 밝혔다. 스마트 임베디드 플랫폼 사업단(SEP) 박강 단장은 “현재 4차산업혁명이 전 세계적으로 일어나고, 더 빠른 속도로 우리

기사를 긁어오는 것은 불법은 아니다. 그런데 자기 기사처럼 쓰는 것은 불법이다.  
그리고 중간중간 sleep을 넣어줘야지 이렇게 긁어오는건 예의는 아니다. 사람이 손으로 클릭하듯이 천천히 긁어오는 방법도 있다.  

* 중앙일보의 기사를 스크롤링하는 내용 총정리

1. 검색 키워드(예: 인공지능)를 가지고 검색을 한 후 그 url을 얻어낸다.
2. 상세기사 url을 리스트에 담는 j_scroll()함수를 생성한다.
3. 상세기사 url로 기사본문을 스크롤링하는 j_detail_scroll() 함수를 생성한다.
4. j_detail_scroll() 함수 안에서 j_scroll() 함수를 호출해서 상세기사 url 여러개를 받아오게끔 코딩하고 받아온 상세기사 url로 본문 기사들을 params2에 담게끔 코드를 수정한다.  

문제417. 동아일보에서 검색 키워드(예: 인공지능)를 가지고 검색을 한 후 그 url을 얻어낸다.
```
https://www.donga.com/news/search?p=1&query=%EC%9D%B8%EA%B3%B5%EC%A7%80%EB%8A%A5&check_news=1&more=1&sorting=1&search_date=1&v1=&v2=&range=1
```


문제418. 상세기사 url을 리스트에 담는 d_scroll() 함수를 생성하세요~

In [ ]:
def d_scroll(page_num): # 중앙일보 페이지 받아옴
    #1. 웹스크롤링에 필요한 모듈을 임포트합니다.
    from bs4 import BeautifulSoup
    import urllib.request

    #2. 중앙일보에서 인공지능으로 검색했을 때 나오는 첫페이지의 html 코드를
    # beautifulSoup에서 이용할 수 있도록 파싱합니다.
    list_url = f"https://www.donga.com/news/search?p={page_num}&query=%EC%9D%B8%EA%B3%B5%EC%A7%80%EB%8A%A5&check_news=1&more=1&sorting=1&search_date=1&v1=&v2=&range=1"
    url = urllib.request.Request(list_url)
    result = urllib.request.urlopen(url).read().decode("utf-8")
    soup = BeautifulSoup(result, "html.parser")

    #3. 상세기사 url을 가져올 수 있도록 태그와 클래스를 찾습니다.
    # 찾아보니 태그는 h2였고 클래스 이름은 headline mg 입니다.
    result1 = soup.find_all('p',class_ ='txt') # find랑 find_all 비교

    #4. 위의 result1은 리스트 이므로 for loop 문을 이용해서 리스트에 있는 요소를
    # 하나씩 빼내면서 상세 url에 해당하는 href의 값을 얻어냅니다.
    params = []
    for i in result1:   # result1 리스트의 요소를 하나씩 빼내는 코드
        for k in i:     # h2 태그안의 a 태그의 html 코드를 가져오기 위함
            params.append(k.get('href'))    # 이거 기억하자
    return params

tit는 이미지 링크고 txt는 텍스트 링크라서 tit하면 에러가 걸린다.

In [ ]:
d_scroll(1)

['https://www.donga.com/news/article/all/20201216/104475349/1',
 'https://www.donga.com/news/article/all/20201216/104472426/2',
 'https://www.donga.com/news/article/all/20201216/104471612/1',
 'https://www.donga.com/news/article/all/20201216/104471110/1',
 'https://www.donga.com/news/article/all/20201215/104463480/1',
 'https://www.donga.com/news/article/all/20201215/104463391/1',
 'https://www.donga.com/news/article/all/20201215/104449354/1',
 'https://www.donga.com/news/article/all/20201215/104461113/1',
 'https://www.donga.com/news/article/all/20201215/104459002/1',
 'https://www.donga.com/news/article/all/20201215/104454507/1',
 'https://www.donga.com/news/article/all/20201215/104448246/1',
 'https://www.donga.com/news/article/all/20201215/104448168/2',
 'https://www.donga.com/news/article/all/20201215/104447953/2',
 'https://www.donga.com/news/article/all/20201214/104445042/1',
 'https://www.donga.com/news/article/all/20201214/104444592/1']

문제419. 상세기사 url로 기사 본문을 스크롤링하는 d_detail_scroll() 함수를 생성한다.

In [ ]:
def d_detail_scroll(num): # url 주소 받아옴
    list_url = d_scroll(num)

    params2 = []
    for i in list_url:
        url = urllib.request.Request(i)
        result = urllib.request.urlopen(url).read().decode("UTF-8")
        soup = BeautifulSoup(result,"html.parser")

        result1 = soup.find_all('div',class_ = "article_txt")

        for i in result1: # result1 리스트의 요소를 하나씩 빼내는 코드
            params2.append(i.get_text(" ", strip = True))
    return params2

In [ ]:
d_detail_scroll(1)

문제420. 한겨레 신문사에서 인공지능으로 검색했을 때 나오는 기사 본문을 스크롤링하는 함수 두개를 생성하시오! 
```
h_scroll() : 상세기사 url을 가져오는 함수
h_detail_scroll() : 상세기사 url로 기사 본문을 가져오는 함수
```


In [ ]:
def h_scroll(page_num): # 중앙일보 페이지 받아옴
    #1. 웹스크롤링에 필요한 모듈을 임포트합니다.
    from bs4 import BeautifulSoup
    import urllib.request
    page_num -= 1
    #2. 중앙일보에서 인공지능으로 검색했을 때 나오는 첫페이지의 html 코드를
    # beautifulSoup에서 이용할 수 있도록 파싱합니다.
    list_url = f"http://search.hani.co.kr/Search?command=query&keyword=%EC%9D%B8%EA%B3%B5%EC%A7%80%EB%8A%A5&media=news&submedia=&sort=d&period=all&datefrom=1988.01.01&dateto=2020.12.16&pageseq={page_num}"
    url = urllib.request.Request(list_url)
    result = urllib.request.urlopen(url).read().decode("utf-8")
    soup = BeautifulSoup(result, "html.parser")

    #3. 상세기사 url을 가져올 수 있도록 태그와 클래스를 찾습니다.
    # 찾아보니 태그는 h2였고 클래스 이름은 headline mg 입니다.
    result1 = soup.find_all('dt') # 클래스 이름 쓰지 말고 dt만

    #4. 위의 result1은 리스트 이므로 for loop 문을 이용해서 리스트에 있는 요소를
    # 하나씩 빼내면서 상세 url에 해당하는 href의 값을 얻어냅니다.
    params = []
    for i in result1:   # result1 리스트의 요소를 하나씩 빼내는 코드
        for k in i:     # h2 태그안의 a 태그의 html 코드를 가져오기 위함
            params.append('http:'+k.get('href'))    # 이거 기억하자
    return params

In [ ]:
h_scroll(1)

['http://www.hani.co.kr/arti/society/ngo/974369.html',
 'http://www.hani.co.kr/arti/opinion/column/974362.html',
 'http://www.hani.co.kr/arti/society/media/974346.html',
 'http://www.hani.co.kr/arti/economy/economy_general/974292.html',
 'http://www.hani.co.kr/arti/economy/economy_general/974274.html',
 'http://www.hani.co.kr/arti/international/international_general/974227.html',
 'http://www.hani.co.kr/arti/society/ngo/974199.html',
 'http://www.hani.co.kr/arti/economy/economy_general/974056.html',
 'http://www.hani.co.kr/arti/international/international_general/973898.html',
 'http://www.hani.co.kr/arti/economy/economy_general/973858.html']

In [ ]:
def h_detail_scroll(num): # url 주소 받아옴
    list_url = h_scroll(num)

    params2 = []
    for i in list_url:
        url = urllib.request.Request(i)
        result = urllib.request.urlopen(url).read().decode("UTF-8")
        soup = BeautifulSoup(result,"html.parser")

        result1 = soup.find_all('div',class_ = "text")

        for i in result1: # result1 리스트의 요소를 하나씩 빼내는 코드
            time.sleep(2)
            params2.append(i.get_text(" ", strip = True))
    return params2

In [ ]:
h_detail_scroll(1)

["◇ 육군은 15일 국립대전현충에서 ‘6·25 전사자 발굴 유해 합동 안장식'을 엄수했다. 안장된 전사자는 고 배석래 이등중사, 송해경 이등중사, 이형술 하사, 문장춘 일병, 명한협 일병, 김학제 일병이다. ◇ 국립광주박물관·임방울국악진흥회는 19일 ‘국창 임방울의 소리향연'을 한다. 김천농악단의 오형우 외 6명의 풍물 판굿 무대, 부채춤, 판소리, 퓨전국악 등을 선보인다. 오프라인 관람은 누리집(gwangju.museum.go.kr)에서 사전 예약을 하면 되고, 유튜브 채널(youtube.com/channel/UCyZbaQ52Gh8SBr6GDdYx-Bw)에서 생중계한다. ◇ 한국언론진흥재단은 16일 언론의 신뢰 회복과 저널리즘 구현을 위한 ‘언론윤리헌장(안)' 제정 공청회를 온라인으로 한다. 언론윤리헌장(안)의 제정 경위와 경과보고, 주요 내용 설명, 언론 3단체 입장 발표 등이 진행되며, 언론진흥재단 유튜브 채널(youtube.com/channel/UCK9lNxYzWToTDhozDv6MSlA)에서 볼 수 있다. ◇ 넥슨재단은 푸르메재단 넥슨어린이재활병원에 병원 내 감염관리 체계 강화를 위한 운영 기금 3억820만원을 기부했다. 기금에는 지난달 열린 넥슨 콘텐츠 축제 ‘네코제9'의 판매 수익금이 포함됐다. ◇ 과학기술정보통신부(장관 최기영)는 16일 ‘2020 ICT 기기 산업 페스티벌'을 온라인으로 한다. 인공지능 기술이 적용된 컴퓨팅 서버 등 국내에서 개발된 신기술과 제품을 선보이는 전시회와 기술 동향과 시장 발전 방향을 논의하는 컨퍼런스를 한다. 누리집(ictfestival.kr)에서 사전등록 후 무료로 볼 수 있다. ◇ 중소기업중앙회(회장 김기문)는 15일 소기업·소상공인을 위한 공적 공제 제도 ‘노란우산'의 홍보대사로 배우 김영철과 가수 송가인을 위촉했다. ◇ 행정안전부 대구청사관리소(소장 이상수)는 15일 인사혁신처가 주관한 ‘2020년 공무원 재능나눔·자원봉사 우수사례'에서 최우수기관으로 선정됐다. 2016년부터 관내 소외계층을 대상으로 진행해

문제421. 오늘의 마지막 문제) 신문사 하고싶은 곳에서 스크롤링 하고싶은 키워드를 넣고 검색했으 ㄹ때 나오는 본문기사들을 수집하는 함수 2개를 생성하시오 ~  
(밑에 페이지 번호 1,2,3 만 스크롤링)
1. jj_scroll() : 상세기사 url 가져오는 함수
2. jj_detail_scroll() : 상세기사 url로 본문 기사를 가져오는 함수
```
https://search.naver.com/search.naver?&where=news&query=%EC%9D%B8%EA%B3%B5%EC%A7%80%EB%8A%A5&sm=tab_pge&sort=0&photo=0&field=0&reporter_article=&pd=0&ds=&de=&docid=&nso=so:r,p:all,a:all&mynews=0&cluster_rank=50&start=1&refresh_start=0
```

In [ ]:
def jj_scroll(page_num):
    from bs4 import BeautifulSoup
    import urllib.request

    list_url = f"https://search.etnews.com/etnews/search.php?category=CATEGORY1&kwd=%EC%9D%B8%EA%B3%B5%EC%A7%80%EB%8A%A5&pageNum={page_num}&pageSize=10&reSrchFlag=false&sort=1&startDate=&endDate=&sitegubun=&jisikgubun=&preKwd%5B0%5D=%EC%9D%B8%EA%B3%B5%EC%A7%80%EB%8A%A5"
    url = urllib.request.Request(list_url)
    result = urllib.request.urlopen(url).read().decode("UTF-8")
    soup = BeautifulSoup(result,"html.parser")

    result1 = soup.find_all('dd',class_ = "summury")
    params = []

    for i in result1:   # result1 리스트의 요소를 하나씩 빼내는 코드
        for k in i:     # h2 태그안의 a 태그의 html 코드를 가져오기 위함
            params.append('http:' + k.get('href'))    # 이거 기억하자
    return params


In [ ]:
jj_scroll(1)

['http://www.etnews.com/20201216000279',
 'http://www.etnews.com/20201216000272',
 'http://www.etnews.com/20201216000184',
 'http://www.etnews.com/20201216000202',
 'http://www.etnews.com/20201216000207',
 'http://www.etnews.com/20201216000256',
 'http://www.etnews.com/20201216000223',
 'http://www.etnews.com/20201216000253',
 'http://www.etnews.com/20201216000227',
 'http://www.etnews.com/20201216000211']

In [ ]:
def jj_detail_scroll(num): # url 주소 받아옴
    list_url = jj_scroll(num)

    params2 = []
    for i in list_url:
        url = urllib.request.Request(i)
        result = urllib.request.urlopen(url).read().decode("UTF-8")
        soup = BeautifulSoup(result,"html.parser")

        result1 = soup.find_all('p')
        print(result1)
        for i in result1: # result1 리스트의 요소를 하나씩 빼내는 코드
 #           time.sleep(2)
            params2.append(i.get_text(" ", strip = True))
    return params2

In [ ]:
num = 1
list_url = jj_scroll(num)

params2 = []
for i in list_url:
    url = urllib.request.Request(i)
    result = urllib.request.urlopen(url).read().decode("UTF-8")
    soup = BeautifulSoup(result,"html.parser")

    result1 = soup.find_all('p' == 0)
    
    for i in result1: # result1 리스트의 요소를 하나씩 빼내는 코드
#           time.sleep(2)
        params2.append(i.get_text(" ", strip = True))
        print(i.get_text(" ", strip = True))

스트리밍 출력 내용이 길어서 마지막 5000줄이 삭제되었습니다.
SW&게임&성장기업
SW&게임&성장기업
소재&부품
소재&부품
전자&자동차&유통
전자&자동차&유통
경제&금융
경제&금융
산업&과학&정책
산업&과학&정책
전국
전국
골프
골프
화제의뉴스
화제의뉴스
인사·부음
인사·부음
오피니언
오피니언
특집
특집
연재
연재

if (navigator.userAgent.match(/(iPod|iPhone|iPad|Android)/) != null) { $('#wingBanner').hide(); } var clean_expire = 0;

			function login_chk(){
				alert('로그인 후 이용해주세요.');
				location.href='https://member.etnews.com/member/login.html?return_url=//www.etnews.com/news/article.html?id=20201216000272';
			} 경기도, '2020 경기 데이터 데이' 행사 개최 발행일 : 2020.12.16 글자 작게 글자 크게 인쇄하기 <임문영 경기도 미래성장정책관이 2020 경기 데이터 데이 행사에서 인사말을 하고 있다.> 교통량과 교통수단 패턴의 변화 데이터로 코로나19 확진자수를 예측한 연구가 경기도의 '데이터를 활용한 코로나19 확진자 추정 모델링 공모전'에서 최우수상에 선정됐다. 경기도와 경기도경제과학진흥원은 16일 현재까지의 도 데이터 정책 성과를 공유하고 앞으로 정책 방향을 알리기 위한 취지로 '2020 경기 데이터 데이' 행사를 온라인 생중계로 진행했다. 행사는 데이터를 활용한 코로나19 확진자 추정 모델링 공모전 시상, 경기도 데이터 정책 활성화 유공자 표창에 이어 경기도 데이터 정책 성과보고회, 데이터 정책 및 기술 관련 연사 초청 강연으로 이뤄졌다. 경기도는 지난 7~11월 '추석기간 동안 경기도 내 코로나19 지역감염 확진자 수 추정 및 지역감염을 줄이기 위한 정책 제언'을 주제로 공모전을 진행했으며 예선과

KeyboardInterrupt: ignored

In [ ]:
    result = urllib.request.urlopen(url).read().decode('UTF-8')
    soup = BeautifulSoup(result,"html.parser")
    
    result1 = soup.find_all('div')
    print(result1)
    params = []
    for i in result1:   # result1 리스트의 요소를 하나씩 빼내는 코드
        for k in i:     # h2 태그안의 a 태그의 html 코드를 가져오기 위함
            params.append(k.get('href'))    # 이거 기억하자
    return params


In [ ]:
jj_scroll(2)